drcln

```diff
@@ -36,17 +36,35 @@ if (!exists('opt'))
             make_option(c("-f", "--germline.file"), default = NA, type = "character", help = "Path to file annotated with germline calls, if germline.filter == TRUE"),
             make_option(c("-m", "--human"), type = "logical", default = TRUE, help = "Specify if the samples under consideration are human"),
             make_option(c("-F", "--field"), type = "character", default = 'reads.corrected', help = "Field name in GRanges metadata to use for drycleaning"),
-            make_option(c("-C", "--all.chr"), default = c(as.character(1:22), "X"), type = "character", help = "list of chromosomes to dryclean"),
+            make_option(c("-C", "--all.chr"), default = "1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, X", type = "character", help = "list of chromosomes to dryclean"),
             make_option(c("-B", "--build"), type = "character", default = 'hg19', help = "hg19/hg38 build for human samples"),
-            make_option(c("-o", "--outdir"), type = "character", default = './', help = "output directory")
+            make_option(c("-o", "--outdir"), type = "character", default = './', help = "output directory"),
+            make_option(c("-T", "--testing"), type = "character", default = FALSE, help = "DO NOT CHANGE"),
+            make_option(c("-h", "--help"), action="store_true", default=FALSE, help="Show this help message and exit"),
+            make_option(c("--mode"), type = "character", default = "coverage", help = "Mode of operation: 'pon' or 'coverage'"),
+            make_option(c("--normal_vector"), type = "character", default = NULL, help = "if create_new_pon = TRUE, vector containing normal coverages. See manual for details"),
+            make_option(c("--field_pon"), type = "character", default = 'reads.corrected', help = "Field to use for PON generation processing")
 
         )
 
         parseobj = OptionParser(option_list=option_list, add_help_option=FALSE)
         opt = parse_args(parseobj)
-
-        if ((is.null(opt$input)) | is.null(opt$pon))
-            stop(print_help(parseobj))
+        
+        if(opt$mode != "coverage" & opt$mode != "pon"){
+          stop("Invalid mode: choose 'pon' or 'coverage'")
+        }
+        
+        if(opt$mode == "pon" & is.null(opt$normal_vector)){
+          stop("Use normal_vector to generate a new PON")
+        }
+        
+        if(opt$mode == "coverage" & is.null(opt$input)){
+          stop("Missing coverage to normalize. Run ./drcln -h for help")
+        }
+        
+        if(opt$mode == "coverage" & is.null(opt$pon)){
+          stop("Missing PON. Run ./drcln -h for help")
+        }
```

Comment:
You should probably move this code down below (line 88 and 101) in order to avoid redundancy (checking the mode twice)